### PR TITLE
added autoeval fields to repocard types

### PR DESCRIPTION
**Pull request by TristanThrush**
This PR makes the repocard types consistent with the spec that autoeval uses: https://raw.githubusercontent.com/huggingface/hub-docs/main/modelcard.md

It is based on lewis's PR [here](https://github.com/huggingface/huggingface_hub/pull/884), but it only includes the necessary changes to repocard_types.py

### CODE REVIEW

File: src/huggingface_hub/repocard_types.py
Comment (lewtun) [inline src/huggingface_hub/repocard_types.py:39]: Maybe we can take the occasion to document this field as well? E.g. something like

```suggestion
    args: Any = field(default=None)
    """
    The arguments passed during `Metric.compute()`. Example for `bleu`: max_order: 4
    """
```

File: src/huggingface_hub/repocard_types.py
Comment (lewtun) [inline src/huggingface_hub/repocard_types.py:107]: For reviewers: this ordering of task > dataset > metrics is more natural and easier to review during Hub PRs

**Review by lewtun — APPROVED** [review]
Thanks a lot for adding this - LGTM! 

Let's wait for a review from the `huggingface_hub` maintainers before merging this. 

cc @LysandreJik @julien-c

**Review by julien-c — COMMENTED** [review]
looks good to me!

**Comment by adrinjalali** [comment]
cc @nateraw

File: tests/test_repocard.py
Comment (nateraw) [inline tests/test_repocard.py:107]: nit - should we add `config` in example here too, since you added that with this PR?

**Review by nateraw — APPROVED** [review]
Might be nice to show how to use this in one of the docstrings in `repocard.py`, but other than that looks good to me.

**Comment by lewtun** [comment]
Hi @adrinjalali can you please advise us on how to handle the failing `codecov` test? Is it a question of writing more unit tests for the repocard or something else?

**Comment by lewtun** [comment]
In order to migrate the AutoTrain backend to use #888 instead of #884, we'll actually need the changes in this PR to be available in @SBrandeis' [fork/branch](https://github.com/huggingface/moon-landing/pull/3144). (We rely on the additional metadata fields provided by the current PR)

To unblock @coyotte508 in https://github.com/huggingface/moon-landing/pull/3144 (which disables the `upload_file()` endpoint we use in AutoTrain) one possibility is to:

1. Close #901 and open an identical one on @SBrandeis's [branch](https://github.com/huggingface/moon-landing/pull/3144) that takes into account the reviewer feedback.
2. Migrate the AutoTrain backend to use #888 once step (1) is merged

This isn't a pretty solution and it pollutes #888 with changes unrelated to the new API. 

An alternative would be to:

* Allow https://github.com/huggingface/moon-landing/pull/3144 to disable the `upload_file()` endpoint we use in AutoTrain. This will disable the evaluation service until #888 is merged 
* Merge #901
* Migrate the AutoTrain backend to the new commit API once #888 is merged

In the interest of releasing a stable evaluation service without lots of hacks, I'm in favour of the second option as long as it doesn't delay the release by too much. @SBrandeis do you have a rough idea how long it will take to get your PR merged?

cc @douwekiela @TristanThrush @abhishekkrthakur

**Comment by lewtun** [comment]
After discussing internally, we'll go for the option to wait until #888 is merged before migrating the AutoTrain backend. 

@coyotte508 from my side: feel free to merge https://github.com/huggingface/moon-landing/pull/3144 when you want :)

**Comment by coyotte508** [comment]
@lewtun https://github.com/huggingface/moon-landing/pull/3144 is ready to be merged

Nothing's going to burn, right? :sweat_smile:

File: tests/test_repocard.py
Comment (TristanThrush) [reply tests/test_repocard.py:107]: Thanks for this feedback!

In order to do this, were you thinking that I should also update the inputs to metadata_eval_result function here: https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/repocard.py#L80

An additional change like this is needed, because the test here won't pass otherwise:
https://github.com/huggingface/huggingface_hub/blob/main/tests/test_repocard.py#L166

**Comment by lewtun** [comment]
> Nothing's going to burn on your side, right? 😅

Nope, the eval service is still in dev but let's quickly ask @gante for confirmation since I believe he was using my branch in `transformers`

**Comment by gante** [comment]
Don't mind `transformers`, I'm the sole user of the related functionality and I can hold it for a few days.

I will make the appropriate changes as soon as #888 gets merged 👍

File: tests/test_repocard.py
Comment (TristanThrush) [reply tests/test_repocard.py:107]: I think that's what you were thinking. I just pushed a change where I added optional fields to metadata_eval_result and added documentation. lmk what you think!

**Review by adrinjalali — COMMENTED** [review]
> Hi @adrinjalali can you please advise us on how to handle the failing codecov test? Is it a question of writing more unit tests for the repocard or something else?

`codecov` runs two tests, one to check the percentage of tests in the PR's patch, the other one on the whole project. For this PR, the only thing left to test now are `metrics_config` and `metrics_verified`.

File: src/huggingface_hub/repocard.py
Comment (TristanThrush) [reply src/huggingface_hub/repocard.py:133]: Thanks, added setting of metrics_config and metrics_verified to tests and did the reordering suggestion that you mentioned for the documentation.

**Pull request merged**
